tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs

```diff
@@ -56,14 +56,23 @@ internal static partial class HttpRequestExtensions
             foreach (var originalKey in request.QueryString.AllKeys)
             {
                 var values = request.QueryString.GetValues(originalKey);
-                var keyForDictionary = originalKey ?? string.Empty;
-                if (!queryDic.ContainsKey(keyForDictionary))
+                if (string.IsNullOrEmpty(originalKey))
+                {
+                    foreach (var v in values)
+                    {
+                        if (!queryDic.ContainsKey(v))
+                        {
+                            queryDic.Add(v, new string[0]);
+                        }
```

Comment:
Can't you do that instead?
```suggestion
                        queryDic.TryAdd(v, Array.Empty<string>());
```

Reply:
The `TryAdd` method is only present in .NET core, this code targets .NET Framework, so the method isn't available. I'll make the change `new string[0]` => `Array.Empty<string>()` in an up coming PR, to avoid a CI run, just for this.

Reply:
:shakesfist: I fall into this trap each time!